codecs/avif/enc/avif_enc.cpp

```diff
@@ -81,23 +81,16 @@ val encode(std::string buffer, int width, int height, AvifOptions options) {
   avifEncoder* encoder = avifEncoderCreate();
 
   if (lossless) {
-    encoder->minQuantizer = AVIF_QUANTIZER_LOSSLESS;
-    encoder->maxQuantizer = AVIF_QUANTIZER_LOSSLESS;
-    encoder->minQuantizerAlpha = AVIF_QUANTIZER_LOSSLESS;
-    encoder->maxQuantizerAlpha = AVIF_QUANTIZER_LOSSLESS;
+    encoder->quality = AVIF_QUALITY_LOSSLESS;
+    encoder->qualityAlpha = AVIF_QUALITY_LOSSLESS;
   } else {
-    encoder->minQuantizer = AVIF_QUANTIZER_BEST_QUALITY;
-    encoder->maxQuantizer = AVIF_QUANTIZER_WORST_QUALITY;
-    encoder->minQuantizerAlpha = AVIF_QUANTIZER_BEST_QUALITY;
-    encoder->maxQuantizerAlpha = AVIF_QUANTIZER_WORST_QUALITY;
     avifEncoderSetCodecSpecificOption(encoder, "end-usage", "q");
-    avifEncoderSetCodecSpecificOption(encoder, "cq-level", std::to_string(options.cqLevel).c_str());
     avifEncoderSetCodecSpecificOption(encoder, "sharpness",
                                       std::to_string(options.sharpness).c_str());
 
+    encoder->quality = options.cqLevel;
```

Comment:
At line 88, check the return value of `avifEncoderSetCodecSpecificOption()`.

Reply:
Sure, and on a related note is there any documentation for the possible key-value pairs for `avifEncoderSetCodecSpecificOption`? The inline documentation says:

> See the codec documentation to know if a setting is persistent or applied only to the next frame.

but I couldn't find any references.

Reply:
Here "codec" refers to the AV1 codec that libavif is configured to use. In Squoosh, libavif is configured to use only libaom as the AV1 codec. So "See the codec documentation" means "See the libaom documentation".

Now comes the hard part. The libaom documntation for the possible key-value pairs for `avifEncoderSetCodecSpecificOption` is best obtained in one of two ways:
- Read the source code in the libaom source file av1/arg_defs.c. This is the definitive way.
- Run the libaom command-line program `aomenc --help`. This is more "user friendly", but has the drawback that `aomenc` has some additional command-line options that are not implemented using libaom's key-value API, so not everything in the `aomenc --help` can be passed to `avifEncoderSetCodecSpecificOption`.

My apologies for the current state of libaom's documentation for key-value pairs. Do you have any specific questions about them? I will be happy to find the answers for you (and tell you how I find the answers so you will be able to do it yourself in the future).

Reply:
No specific questions so far, though I'm sure I'll have some as I dive deeper into the codec. [The libaom file has pretty good documentation](https://aomedia.googlesource.com/aom/+/refs/heads/main/av1/arg_defs.c#145) so that answers the question!